### DISCUSSION TITLE
[Data Indexing] Known issue relating to indexing of internal transaction traces ("Awaiting internal transactions for status")

### DISCUSSION THREAD

**Discussion by arthurgousset** [celo-org/blockscout#916] [Q&A] [answered]
## TLDR

> **Note**
> + A temporary issue is affecting data services of `explorer.celo.org` such as the [website](https://explorer.celo.org/), [HTTP API](https://explorer.celo.org/mainnet/api-docs), and [GraphQL API](https://explorer.celo.org/mainnet/graphiql).
> + The Celo blockchain is working correctly, your funds are safe, and transactions are confirmed on the blockchain as expected.
> + The issue relates to **data indexing** on our side and is restricted to `explorer.celo.org` services. As such it does not affect other data infrastructure providers such as [celoscan.io](https://celoscan.io/), [QuickNode](https://www.quicknode.com/), [Infura](https://www.infura.io/), [TheGraph](https://thegraph.com/), etc.
> + During the incident, we recommend you use [celoscan.io](https://celoscan.io/) to double-check recent transactions and blocks on Celo mainnet and Alfajores testnet.

We apologise for the inconvenience and are working hard to bring the service back. Thank you for your patience! 🙏

## More details

On Friday June 9th, the indexing engine that powers the `explorer.celo.org` [website](https://explorer.celo.org/), [HTTP API](https://explorer.celo.org/mainnet/api-docs), and [GraphQL API](https://explorer.celo.org/mainnet/graphiql) started experiencing performance issues. 

Specifically, our indexing engine started falling behind on **internal transaction traces**, which are expensive to compute, but not **ordinary transactions**, which are cheaper to compute. 

This means that:

1. **ordinary transactions** are indexed in time, specifically, as soon as the latest block is confirmed on the blockchain, and the respective transaction data available via the `explorer.celo.org` website and APIs is up-to-date.
2. **internal transaction traces** currently take longer to compute, which means that internal transaction data is delayed and temporarily missing in the `explorer.celo.org` website and APIs while our indexing engine is experiencing performance issues.

The symptoms of this issue manifest the following way:

- **Transaction result** message: `(Awaiting internal transactions for status)`
  - Example: [explorer.celo.org/baklava/tx/0x62e25558aa6574ce...](https://explorer.celo.org/baklava/tx/0x62e25558aa6574ce6d66ce6b0250844addbd0c33d454b5d4be9f98dda694cc33/internal-transactions)
  - Notice that the transaction is confirmed, but the transaction result is pending due to the internal transaction indexing issue.
    <img width="600" alt="image" src="https://github.com/celo-org/blockscout/assets/46296830/9d3f09f0-a7db-47d6-9701-906845a88c22">

You can follow the fix for this issue in this pull request:

- https://github.com/celo-org/blockscout/pull/918

This issue is being tracked here:

- https://github.com/celo-org/blockscout/discussions/917

**Comment by arthurgousset** [comment]
## Bug reports by the community (raw)

Replicating a few bugs reported on Slack

### Contract Call (`Awaiting internal transactions for status`)

Reported by: @ericnakagawa and @nestorbonilla

> Also, is “Contract Call (Awaiting internal transactions for status)” an acceptable status?
> 
> <img src="https://github.com/celo-org/blockscout/assets/46296830/d1f4a9d8-0cfc-4885-963a-6da9209e65aa" width="850">
> 
> https://explorer.celo.org/baklava/tx/0x62e25558aa6574ce6d66ce6b0250844addbd0c33d454b5d4be9f98dda694cc33
> 
> <img src="https://github.com/celo-org/blockscout/assets/46296830/b7b3c6f5-834f-4323-a7f1-e1b88340dfc3" width="850">

### Contract Call (`Awaiting internal transactions for status`)

Reported by: @martinvol

> i’m seeing a couple of tx in Baklava that show `(Awaiting internal transactions for status)`, the tx also took quite a while to show up as confirmed in the explorer.

### Balance inaccurate in blockchain explorer

Reported by: @ericnakagawa and @nestorbonilla

> hey all Nestor Bonilla and I are encountering issues with the blockchain explorer being inaccurate.
> 
> https://explorer.celo.org/baklava/address/0x60e5d9041620C7728fF0970454F09A95d2C44dCc
> balance in drop down shows 22000 cGLD, but balance on main section shows 11989… CELO
> <img src="https://github.com/celo-org/blockscout/assets/46296830/ab8c7121-bac5-4ab7-829b-7ef08fdadea8" width="450">

**Comment by yerdua** [comment, marked as answer]
### Update - incident resolved

As of today (Tuesday, July 12), indexing has caught up with internal transaction tracing and is once again operating normally. We consider this issue resolved, but will continue to monitor its performance.
